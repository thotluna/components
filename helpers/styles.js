import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
