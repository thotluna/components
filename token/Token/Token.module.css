.container {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
}

.description {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: 10px;
  gap: 4px;
}

.label {
  min-width: max-content;
  padding: 4px 8px;
  border: 1px solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
  border-radius: var(--border-radius-sm);
  color: var(--color-font-base);
  font-family: var(--font-family-mono);
  font-size: var(--font-size-sm);
}

.label:first-child {
  border: var(--border-width-thin) solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
}

.color {
  width: 65px;
  height: 65px;
  border: var(--button-base-color-border);
  border-radius: var(--border-radius-full);
  box-shadow: var(--box-shadow-sm);
}

.spacer {
  display: flex;
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
}

.font-size,
.font-family {
  width: 100%;
  max-width: 500px;
  text-align: justify;
}

.font-size {
  text-align: right;
}

.title {
  font-family: var(--font-family-title);
  font-size: 16px;
}

.sans {
  font-family: var(--font-family-sans);
  font-size: 16px;
}

.box {
  width: 20rem;
  height: 5rem;
  border: 1px solid black;
}
