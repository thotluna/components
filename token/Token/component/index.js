export { default as Color } from './Colors/Colors'
export { default as Spacing } from './Spacing/Spacing'
export { default as FontFamily } from './Fonts/FontFamily'
export { default as FontSize } from './Fonts/FontSize'
export { default as FontWeight } from './Fonts/FontWeight'
