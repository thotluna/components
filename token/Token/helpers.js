import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
