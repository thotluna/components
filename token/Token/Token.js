import React from 'react'

import { Color, Spacing, FontFamily, FontSize, FontWeight } from './component'

const Token = () => <></>

Token.Color = Color
Token.Spacing = Spacing
Token.FontFamily = FontFamily
Token.FontSize = FontSize
Token.FontWeight = FontWeight

export default Token
