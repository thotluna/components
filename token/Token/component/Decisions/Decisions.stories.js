import Token from '../..'

import { decisions } from '../../..'

import {
  getTemplate,
  getListTemplate,
  getVariusTemplate,
} from '../../../../helpers/storybook'
import { getTokenItems } from '../../helpers'

const Template = getTemplate(Token.Color)
const ListTemplate = getListTemplate(Token.Color)

export default {
  title: 'Tokens/Decisions/Color',
  component: Token.Color,
  argTypes: {
    value: { control: null },
    label: { control: null },
  },
  parameters: { __sb: { mh: 250 } },
}

export const Default = Template.bind({})
Default.args = {
  label: '--color-primary',
  value: decisions.color.primary,
}

const ListTemplateBackground = getVariusTemplate(Token.Color)
const backgroundColor = [
  { label: '--color-dark-background', value: decisions.color.dark.background },
  {
    label: '--color-light-background',
    value: decisions.color.light.background,
  },
]

export const BackgroundColor = ListTemplateBackground.bind({})
BackgroundColor.args = {
  __sb: { fd: 'row' },
  items: backgroundColor,
}
BackgroundColor.parameters = {
  __sb: { fd: 'row' },
}

const getDecisionLightItem = getTokenItems(decisions.color.light, 'color')

export const FontColorWithBackgroundLight = ListTemplate.bind({})
FontColorWithBackgroundLight.args = {
  __sb: { fd: 'row', bc: decisions.color.light.background },
  items: getDecisionLightItem('font'),
}
FontColorWithBackgroundLight.parameters = {
  __sb: { fd: 'row', bc: decisions.color.light.background },
}

const getDecisionDarkItem = getTokenItems(decisions.color.dark, 'color')

export const FontColorWithBackgroundDark = ListTemplate.bind({})
FontColorWithBackgroundDark.args = {
  __sb: { fd: 'row', bc: decisions.color.dark.background },
  items: getDecisionDarkItem('font'),
}
FontColorWithBackgroundDark.parameters = {
  __sb: { fd: 'row', bc: decisions.color.dark.background },
}

export const ButtonColorWithBackgroundLight = ListTemplate.bind({})
ButtonColorWithBackgroundLight.args = {
  __sb: { fd: 'row', bc: decisions.color.light.background },
  items: getDecisionLightItem('button'),
}
ButtonColorWithBackgroundLight.parameters = {
  __sb: { fd: 'row', bc: decisions.color.light.background },
}

export const ButtonColorWithBackgroundDark = ListTemplate.bind({})
ButtonColorWithBackgroundDark.args = {
  __sb: { fd: 'row', bc: decisions.color.dark.background },
  items: getDecisionDarkItem('button'),
}
ButtonColorWithBackgroundDark.parameters = {
  __sb: { fd: 'row', bc: decisions.color.dark.background },
}
