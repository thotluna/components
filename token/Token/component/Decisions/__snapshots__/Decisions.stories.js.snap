// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tokens/Decisions/Color Background Color 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#2B2351",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-dark-background
      </span>
      <span
        className="label"
      >
        #2B2351
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-light-background
      </span>
      <span
        className="label"
      >
        #F5F5F5
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Decisions/Color Button Color With Background Dark 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "#2B2351",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#716181",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-primary
      </span>
      <span
        className="label"
      >
        #716181
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#C91280",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-secondary
      </span>
      <span
        className="label"
      >
        #C91280
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-base
      </span>
      <span
        className="label"
      >
        transparent
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Decisions/Color Button Color With Background Light 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "#F5F5F5",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#716181",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-primary
      </span>
      <span
        className="label"
      >
        #716181
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#C91280",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-secondary
      </span>
      <span
        className="label"
      >
        #C91280
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-button-base
      </span>
      <span
        className="label"
      >
        transparent
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Decisions/Color Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-primary
      </span>
      <span
        className="label"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Decisions/Color Font Color With Background Dark 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "#2B2351",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-font-base
      </span>
      <span
        className="label"
      >
        #fff
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#EDEDED",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-font-muted
      </span>
      <span
        className="label"
      >
        #EDEDED
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Decisions/Color Font Color With Background Light 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "#F5F5F5",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#000",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-font-base
      </span>
      <span
        className="label"
      >
        #000
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#706080",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-font-muted
      </span>
      <span
        className="label"
      >
        #706080
      </span>
    </div>
  </div>
</div>
`;
